### PR TITLE
Issue 5546:  Fix NPE in DynamicRestApiTest

### LINKED ISSUE #5546: Flaky test: DynamicRestApiTest
**Problem description**
SystemTest failure : DynamicRestApiTest

```
2021-01-18 20:45:22,784 2531 [Time-limited test] ERROR i.p.t.system.SingleJUnitTestRunner - Test listScopes failed with exception. 
java.lang.NullPointerException: password is marked non-null but is null  
at io.pravega.shared.security.auth.DefaultCredentials.<init>(DefaultCredentials.java:33)                                                                                                               at io.pravega.test.system.DynamicRestApiTest.listScopes(DynamicRestApiTest.java:120)                                                                                                                   at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)                                                                                                                      at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)                                                                                                                      at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)                                                                                                                  at java.base/java.lang.reflect.Method.invoke(Unknown Source)      
```

**Problem location**
DynamicRestApiTest

**Suggestions for an improvement**
Fix it.

### PR DESCRIPTION
**Pull request by shrids**
**Change log description**  
Fix NPE in DynamicRestApiTest system test.

**Purpose of the change**  
Fixes #5546 

**What the code does**  
The constructor for `DefaultCredentaials ` follows the pattern ` public DefaultCredentials(@NonNull String password, @NonNull String userName)` 
and the current code 
- does not pass the password but username twice
- and the property is not set for a system test run without auth.
This has been fixed by using the helper method `Utils.buildClientConfig(..)`

**How to verify it**  
The system tests should pass.

### CODE REVIEW

**Comment by shrids** [comment]
System test run is still in progress..

File: test/system/src/test/java/io/pravega/test/system/DynamicRestApiTest.java [hunk 1]
```diff
@@ -114,11 +107,7 @@ public void listScopes() {
         Client client = ClientBuilder.newClient();
         String responseAsString = null;
 
-        ClientConfig clientConfig = ClientConfig.builder()
-                .controllerURI(controllerGRPCUri)
-                .credentials(new DefaultCredentials(Utils.PRAVEGA_PROPERTIES.get("pravega.client.auth.username"),
-                        Utils.PRAVEGA_PROPERTIES.get("pravega.client.auth.username")))
-                .build();
+        ClientConfig clientConfig = Utils.buildClientConfig(controllerGRPCUri);
```
Comment (RaulGracia) [inline test/system/src/test/java/io/pravega/test/system/DynamicRestApiTest.java:110]: Can you elaborate a bit more about the root cause of the issue associated to this PR and the fix provided? Apparently, this change is suggesting that the way of initializing the `ClientConfig` is the problem, but it is not explicitly state anywhere. If this is the case, is the `Utils.buildClientConfig()` method handling cases in which system tests are executed with and without security enabled?

Comment (shrids) [reply test/system/src/test/java/io/pravega/test/system/DynamicRestApiTest.java:110]: yes, `Utils.buildClientConfig()` is the method which is used for handling cases in which system tests are executed with/without auth and tls..

File: test/system/src/test/java/io/pravega/test/system/DynamicRestApiTest.java [hunk 2]
```diff
@@ -114,11 +107,7 @@ public void listScopes() {
         Client client = ClientBuilder.newClient();
         String responseAsString = null;
 
-        ClientConfig clientConfig = ClientConfig.builder()
-                .controllerURI(controllerGRPCUri)
-                .credentials(new DefaultCredentials(Utils.PRAVEGA_PROPERTIES.get("pravega.client.auth.username"),
```
Comment (shrids) [inline test/system/src/test/java/io/pravega/test/system/DynamicRestApiTest.java:119]: The constructor follows the pattern ` public DefaultCredentials(@NonNull String password, @NonNull String userName)` 
and the current code 
- does not pass the password but username twice
- and the property is not set for a system test run without auth.

**Review by RaulGracia — APPROVED** [review]
LGTM, let me know if you see this test passing in system tests.

**Comment by shrids** [comment]
test-pravega-continuous-1/1169 is successful

**Comment by RaulGracia** [comment]
@shrids we may need to cherry pick this one to r0.9 for having green system tests with security enabled.

**Pull request merged**